src/views/ProfileView.vue

```diff
@@ -0,0 +1,322 @@
+        if (schule) composedZuordnung.sskName = schule;
+        if (klasse) composedZuordnung.klasse = klasse;
+        composedZuordnungen.push(composedZuordnung);
+      } else {
+        if (zuordnungen.length === 1 && zuordnungen[0]) composedZuordnungen.push(zuordnungen[0]);
+      }
+    }
+
+    return composedZuordnungen;
+  }
+
+  function createZuordnungsSchoolData(zuordnungen: Zuordnung[]): SchoolData[] {
+    const result: SchoolData[] = [];
+    for (const zuordnung of zuordnungen) {
+      const tempSchoolData: SchoolData = {
+        title: zuordnung.sskName,
+        info: t('profile.yourSchoolAdminsAre'),
+        schoolAdmins: [], // Hierfuer muss ein API-Endpunkt implementiert werden
+        labelAndValues: [{ label: t('profile.school'), value: zuordnung.sskName }],
+      };
+
+      if (zuordnung.klasse) {
+        tempSchoolData.labelAndValues.push({
+          label: t('profile.class'),
+          value: zuordnung.klasse,
+        });
+      }
+
+      tempSchoolData.labelAndValues.push({ label: t('admin.rolle.rolle'), value: zuordnung.rolle });
+
+      if (zuordnung.sskDstNr) {
+        tempSchoolData.labelAndValues.push({
+          label: t('profile.schoolNumber'),
+          labelAbbr: t('profile.schoolNumberAbbr'),
+          value: zuordnung.sskDstNr,
+        });
+      }
+
+      result.push(tempSchoolData);
+    }
+
+    return result;
+  }
+
+  onBeforeMount(async () => {
+    await personInfoStore.initPersonInfo();
+    await personenkontextStore.getPersonenuebersichtById(personInfoStore.personInfo?.person.id ?? '');
+    const personInfo: PersonInfoResponse = personInfoStore.personInfo as PersonInfoResponse;
+    const personenZuordnungen: Zuordnung[] = personenkontextStore.personenuebersicht?.zuordnungen ?? [];
+
+    personalData.value = [
+      {
+        label: 'Vor- und Nachname',
+        value: personInfo.person.name.vorname + ' ' + personInfo.person.name.familiennamen,
+      },
+      { label: 'Benutzername', value: personInfo.person.referrer },
+    ];
+
+    if (personInfo.person.personalnummer) {
+      personalData.value.push({
+        label: t('profile.personalNummer'),
+        labelAbbr: t('profile.personalNummerAbbr'),
+        value: personInfo.person.personalnummer,
+      });
+    }
+    const groupedZuordnungen: Map<string, Zuordnung[]> = groupZuordnungen(
+      personenZuordnungen.map(
+        (z: Zuordnung) =>
+          ({
+            ...z,
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+            sskDstNr: z.sskDstNr?.split('-')[0], // die Klasse wird durch einen Bindestrich an die Schulnummer angehangen. Um nach der Schule zu gruppieren, wird nur die Schulnummer verwendet.
+          }) as Zuordnung,
+      ),
+    );
+
+    const composedZuordnungen: Zuordnung[] = createComposedZuordnungen(groupedZuordnungen);
+    schoolDatas.value = createZuordnungsSchoolData(composedZuordnungen);
+  });
```

Comment:
Ich würde hier vielleicht die personalData bzw. schuleData in eigenen Methoden packen und hier im Hook triggern. Ausserdem eine extra Methode um die Store zu initialisieren:

```
async function initializeStores() {
  const personInfoStore = usePersonInfoStore();
  const personenkontextStore = usePersonenkontextStore();

  await personInfoStore.initPersonInfo();
  await personenkontextStore.getPersonenuebersichtById(personInfoStore.personInfo?.person.id ?? '');
}
```
Dann haben wir am Ende nur sowas:

```
onBeforeMount(async () => {
    await initializeStores();
    setupPersonalData();
    setupSchuleData();
  } 
});
```